test/stringio/test_stringio.rb

```diff
@@ -922,6 +922,16 @@ def test_encoding_write
     assert_equal("abc".encode("utf-32be"), s.string)
   end
 
+  def test_write_str_modify
+    seek_pos = [0, 2, 4, 6]
+    seek_pos.each do |pos|
+      s = StringIO.new('abcd')
+      s.seek(pos)
+      s.write("\xff\xff\xff")
+      refute(s.string.ascii_only?)
```

Comment:
```suggestion
      assert_operator(s.string, :ascii_only?)
```

Reply:
changed to `refute_predicate(s.string, :ascii_only?)`

Reply:
Oh, sorry.
Could you use `assert_not_predicate` instead? I don't want to use `refute_*` as much as possible because `assert_not_` is straightforward for non-native English speaker.